### PR TITLE
Fix the possibility to override the logout request uri (end_session endpoint)

### PR DESCRIPTION
**Pull request by kriszman**
Came across this project after reading an article on Baeldung who linked it for good reason.
It's really amazing and I'm really glad I stumbled across it as it's more than helpful setting up and generally getting into oidc and authentication.
However, I had some issues trying to override the end session endpoint provided by the OP discovery as described in the README. When specifying an alternative url in the application properties (oauth2-logout) this url isn't recognized, so I tried to adapt the corresponding parts. Hope this makes sense, thx.

### CODE REVIEW

File: spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/SpringAddonsOAuth2LogoutRequestUriBuilder.java [hunk 1]
```diff
@@ -30,9 +30,6 @@ public class SpringAddonsOAuth2LogoutRequestUriBuilder implements LogoutRequestU
     @Override
     public Optional<String> getLogoutRequestUri(ClientRegistration clientRegistration, String idToken, Optional<URI> postLogoutUri) {
         final var logoutProps = clientProperties.getLogoutProperties(clientRegistration.getRegistrationId());
-        if (logoutProps.map(OAuth2LogoutProperties::isEnabled).orElse(false)) {
```
Comment (ch4mpy) [inline spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/SpringAddonsOAuth2LogoutRequestUriBuilder.java:33]: I think that this test should be kept: as [documented on the property](https://github.com/ch4mpy/spring-addons/blob/a2010649edf33a6b6bd5b7974f9eb7b1b0df55e9/spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/properties/SpringAddonsOidcClientProperties.java#L212) the aim is to give the opportunity to disable RP-Initiated Logout (keep the user session on the authorization server) and skip directly to the post-logout URI.

File: spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/synchronised/client/SpringAddonsLogoutSuccessHandler.java [hunk 2]
```diff
@@ -81,6 +81,6 @@ protected String determineTargetUrl(HttpServletRequest request, HttpServletRespo
 
     @Override
     public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException, ServletException {
-        this.redirectStrategy.sendRedirect(request, response, determineTargetUrl(request, response));
+        this.redirectStrategy.sendRedirect(request, response, determineTargetUrl(request, response, authentication));
```
Comment (ch4mpy) [inline spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/synchronised/client/SpringAddonsLogoutSuccessHandler.java:84]: :thumbsup:

Comment (ch4mpy) [inline spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/synchronised/client/SpringAddonsLogoutSuccessHandler.java:84]: :thumbsup:

**Review by ch4mpy — CHANGES_REQUESTED** [review]
Thks for this PR.

Please restore the ability to disable RP-Initiated Logout from properties, I think that the missing argument you found in th logout success-handler is enough to fix your issue.

File: spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/SpringAddonsOAuth2LogoutRequestUriBuilder.java (hunk 1 above)
Comment (ch4mpy) [reply spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/SpringAddonsOAuth2LogoutRequestUriBuilder.java:33]: True, it should be negated.

As a side note, I wonder how could the [resource-server_with_ui tutorial](https://github.com/ch4mpy/spring-addons/tree/master/samples/tutorials/resource-server_with_ui) ever work...

**Comment by ch4mpy** [comment]
@kriszman thank you again for your investigations and fix.

I'll release this patch in the next 8-10 hours and will ping you when the new version is available from maven-central.

**Comment by kriszman** [comment]
> @kriszman thank you again for your investigations and fix.
> 
> I'll release this patch in the next 8-10 hours (don't have the required GPG keys at hand) and will ping you when the new version is available from maven-central.

no worries, it's been really a small issue, happy I could assist a little. Thank you for this amazing library, hope it keeps on going... and thx for keeping me posted 👍

**Comment by ch4mpy** [comment]
@kriszman the `7.6.6` is available from maven-central. It contains your fix.

**Comment by ch4mpy** [comment]
@kriszman the version to use is actually `7.6.7`. There was still a small issue with the condition to de-activate RP-Initiated Logout: in case where logout properties are not provided at all, we want to keep RP-Initiated Logout => de-activation "not true"

**Comment by kriszman** [comment]
> @kriszman the version to use is actually `7.6.7`. There was still a small issue with the condition to de-activate RP-Initiated Logout: in case where logout properties are not provided at all, we want to keep RP-Initiated Logout => de-activation "not true"

@ch4mpy I see, seems like I've been too focused on my use case, sorry for the inconvenience and thx for the fast feedback and fix!

**Pull request merged**